src/tactic/derive_fintype.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2020 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+-/
+import logic.basic data.fintype.basic
+
+/-!
+# Derive handler for `fintype` instances
+
+This file introduces a derive handle to automatically generate `fintype`
+instances for structures and inductives.
+-/
```

Comment:
Could you add a few "big picture" lines? What's the 5-line sketch of the algorithm?

Reply:
I added a bit more than 5 lines...

Reply:
Beautiful :)